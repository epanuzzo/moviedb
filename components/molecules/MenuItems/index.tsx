import React from "react";
import Link from "next/link";

type MenuItemProps = {
  url: string;
  text: string;
};

type MenuItemsProps = {
  active: string;
  isMobile?: boolean;
  wrapperClassName?: string;
  menuItems?: MenuItemProps[];
};

const MenuItems: React.FC<MenuItemsProps> = ({
  active,
  isMobile = false,
  wrapperClassName = "",
  menuItems = [
    { url: "/", text: "Home" },
    { url: "/about", text: "About" },
  ],
}) => {
  return (
    <nav
      className={`${
        isMobile
          ? "flex md:hidden flex-col space-y-4 pt-4"
          : "hidden md:flex space-x-4 pl-4"
      } ${wrapperClassName}`}
    >
      {menuItems.map((item: MenuItemProps) => (
        <Link
          href={item.url}
          className={`text-1xl ${active === item.url ? "underline" : ""}`}
          key={item.url}
          passHref
        >
          {item.text}
        </Link>
      ))}
    </nav>
  );
};

export default MenuItems;
