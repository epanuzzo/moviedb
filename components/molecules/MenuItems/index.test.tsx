import React from "react";
import renderer from "react-test-renderer";
import MenuItems from "./index";

describe("MenuItems", () => {
  it("renders component", () => {
    const tree = renderer.create(<MenuItems active="/" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders component with alt props", () => {
    const tree = renderer
      .create(<MenuItems active="/" isMobile wrapperClassName="test-class" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
