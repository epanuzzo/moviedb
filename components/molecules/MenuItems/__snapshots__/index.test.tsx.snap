// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItems renders component 1`] = `
<nav
  className="hidden md:flex space-x-4 pl-4 "
>
  <a
    className="text-1xl underline"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Home
  </a>
  <a
    className="text-1xl "
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    About
  </a>
</nav>
`;

exports[`MenuItems renders component with alt props 1`] = `
<nav
  className="flex md:hidden flex-col space-y-4 pt-4 test-class"
>
  <a
    className="text-1xl underline"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Home
  </a>
  <a
    className="text-1xl "
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    About
  </a>
</nav>
`;
